packages/react-native/src/notifier.js

```diff
@@ -29,61 +37,86 @@ const internalPlugins = [
-        // mutate the options with anything supplied in JS. This will throw
-        Object.keys(jsOpts).forEach(k => { opts[k] = jsOpts[k] })
-      }
-    }
+const CLIENT_METHODS = Object.getOwnPropertyNames(Client.prototype)
 
-    const bugsnag = new Client(opts, schema, internalPlugins, { name, version, url })
+const createClientAsync = async (jsOpts) => {
+  const opts = await loadAsync(NativeClient, version)
+  return _createClient(opts, jsOpts)
+}
 
-    bugsnag._setDelivery(client => delivery(client, NativeClient))
+const createClient = (jsOpts) => {
+  const opts = load(NativeClient, version)
+  return _createClient(opts, jsOpts)
+}
 
-    if (opts.user && opts.user !== opts._originalValues.user) {
-      bugsnag.setUser(opts.user.id, opts.user.email, opts.user.name)
-    }
+const _createClient = (opts, jsOpts) => {
+  if (jsOpts && typeof jsOpts === 'object') {
+    // mutate the options with anything supplied in JS. This will throw
+    Object.keys(jsOpts).forEach(k => { opts[k] = jsOpts[k] })
+  }
 
-    if (opts.context && opts.context !== opts._originalValues.context) {
-      bugsnag.setContext(opts.context)
-    }
+  const bugsnag = new Client(opts, schema, internalPlugins, { name, version, url })
 
-    if (opts.metadata && opts.metadata !== opts._originalValues.metadata) {
-      Object.keys(opts.metadata).forEach(k => bugsnag.addMetadata(k, opts.metadata[k]))
-    }
+  bugsnag._setDelivery(client => delivery(client, NativeClient))
 
-    if (opts.codeBundleId) {
-      NativeClient.updateCodeBundleId(opts.codeBundleId)
-    }
+  if (opts.user && opts.user !== opts._originalValues.user) {
+    bugsnag.setUser(opts.user.id, opts.user.email, opts.user.name)
+  }
+
+  if (opts.context && opts.context !== opts._originalValues.context) {
+    bugsnag.setContext(opts.context)
+  }
+
+  if (opts.metadata && opts.metadata !== opts._originalValues.metadata) {
+    Object.keys(opts.metadata).forEach(k => bugsnag.addMetadata(k, opts.metadata[k]))
+  }
+
+  if (opts.codeBundleId) {
+    NativeClient.updateCodeBundleId(opts.codeBundleId)
+  }
 
-    bugsnag._logger.debug('Loaded!')
+  bugsnag._logger.debug('Loaded!')
 
-    return bugsnag
-  },
+  return bugsnag
+}
+
+const Bugsnag = {
+  _client: null,
   start: (opts) => {
     if (Bugsnag._client) {
       Bugsnag._client._logger.warn('Bugsnag.start() was called more than once. Ignoring.')
       return Bugsnag._client
     }
-    Bugsnag._client = Bugsnag.createClient(opts)
-    Bugsnag._client._depth += 1
-    return Bugsnag._client
+    if (!isDebuggingRemotely) {
+      Bugsnag._client = createClient(opts)
+      Bugsnag._client._depth += 1
+      return Bugsnag._client
+    } else {
+      console.warn(REMOTE_DEBUGGING_WARNING)
+      let initialised = false
+      const stubClient = {}
+      CLIENT_METHODS.reduce((accum, m) => {
+        stubClient[m] = new Proxy(() => {
+          console.warn(`Bugsnag.${m}() will no-op because Bugsnag has not yet initialised`)
```

Comment:
Consider: `This call to Bugsnag.${m}() is a no-op because remote debugging is enabled and Bugsnag has not yet initialized`